rules/manager.go

```diff
@@ -41,68 +41,65 @@ import (
 const namespace = "prometheus"
 
 var (
-	evalDuration = prometheus.NewSummaryVec(
+	evalDuration = prometheus.NewSummary(
 		prometheus.SummaryOpts{
 			Namespace: namespace,
 			Name:      "rule_evaluation_duration_seconds",
 			Help:      "The duration for a rule to execute.",
 		},
-		[]string{"rule_type"},
 	)
-	evalFailures = prometheus.NewCounterVec(
+	evalFailures = prometheus.NewCounter(
 		prometheus.CounterOpts{
 			Namespace: namespace,
 			Name:      "rule_evaluation_failures_total",
 			Help:      "The total number of rule evaluation failures.",
 		},
-		[]string{"rule_type"},
 	)
-	evalTotal = prometheus.NewCounterVec(
+	evalTotal = prometheus.NewCounter(
 		prometheus.CounterOpts{
 			Namespace: namespace,
 			Name:      "rule_evaluations_total",
 			Help:      "The total number of rule evaluations.",
 		},
-		[]string{"rule_type"},
 	)
 	iterationDuration = prometheus.NewSummary(prometheus.SummaryOpts{
 		Namespace:  namespace,
-		Name:       "evaluator_duration_seconds",
+		Name:       "rule_group_duration_seconds",
 		Help:       "The duration of rule group evaluations.",
 		Objectives: map[float64]float64{0.01: 0.001, 0.05: 0.005, 0.5: 0.05, 0.90: 0.01, 0.99: 0.001},
 	})
 	iterationsMissed = prometheus.NewCounter(prometheus.CounterOpts{
```

Comment:
Does it make sense to put `rule_group` as a label here and in `rule_group_iterations_total`

Reply:
If you did that then you wouldn't be able to measure the total time spent by all rule groups, as they could appear and disappear.

Reply:
Sorry, I actually meant for `rule_group_iterations_missed_total` not `rule_group_duration_seconds`

Reply:
Same argument applies, and you'll be able to detect that per rule group by the last eval being higher than the interval.